### PR TITLE
Add Cosmos (ATOM) support

### LINKED ISSUE #92: Support Cosmos transactions

### PR DESCRIPTION
**Pull request by jamesscaur**
Closes #92. Previous discussion on #107 and #105.

Not sure what all these other commits have been added for - doesn't seem to affect the code, though. Any way I can remove them?

### CODE REVIEW

**Comment by terorie** [comment]
> Not sure what all these other commits have been added for - doesn't seem to affect the code, though. Any way I can remove them?

Yes, you can squash your commits (https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)

**Comment by jamesscaur** [comment]
> Yes, you can squash your commits (https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)

@terorie I did squash, but I believe it made me 'author' a bunch of non-changes as a result to those other 65 commits. I just found out about cherry-pick though, will update...

**Comment by kolya182** [comment]
@jamesscaur Thank you for PR, now Trust users have transactions history

File: platform/cosmos/api.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+func getTransactions(c *gin.Context) {
+	inputTxes, _ := client.GetAddrTxes(c.Param("address"), "inputs")
+	outputTxes, _ := client.GetAddrTxes(c.Param("address"), "outputs")
+
+	normalisedTxes := make([]models.Tx, 0)
+
+	for _, inputTx := range inputTxes {
+		normalisedInputTx := Normalize(&inputTx)
+		normalisedTxes = append(normalisedTxes, normalisedInputTx)
```
[the remark this answers is not in the document]
Comment (jamesscaur) [reply platform/cosmos/api.go]: Thanks - can you clarify the improvement here? Make sure that normalisedTxes (combined inputs and outputs) don't exceed 25?

File: platform/cosmos/client.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+func (c *Client) GetAddrTxes(address string, inOrOut string) (txs []Tx, err error) {
+	var uri string
+
+	if inOrOut == "inputs" {
+		uri = fmt.Sprintf("%s/txs?%s",
+			c.BaseURL,
+			url.Values{
+				"recipient": {address},
+				"page":      {strconv.FormatInt(1, 10)},
+				"limit":     {strconv.FormatInt(models.TxPerPage, 10)},
```
[the remark this answers is not in the document]
Comment (jamesscaur) [reply platform/cosmos/client.go]: Thanks - can you give me an example of how this would look?
```"limit": {strconv.FormatInt(25, 10)},``` perhaps?

**Comment by jamesscaur** [comment]
Thanks for the review @kolya182, have been using Trust for the last year, feels great to have contributed

**Pull request merged**